Feature aerodynamic surrogate storage

ssankalp26 opened the pull request:
These changes now store an aerodynamic surrgoate for each segment and mission when the flag is true. The user needs to manually delete the pickle file when running a single file. The files are stored in the pickle file, to make sure that they cannot be pushed to github and wont mess with any verification tests.

When a payload range computation is called, it forces the store surrogate to be true, and after the payload range computations are done it will go ahead and delete it. This leads to a 500% increase in speed during the payload range computations.

ssankalp26 committed:
storing surrogate data with respect to segment

ssankalp26 committed:
Add surrogate data control and cleanup for aerodynamics

Introduces a 'store_surrogate_data' setting to aerodynamics analysis and updates payload range diagram computation to enable this setting and remove generated .pkl files after execution. Aerodynamics pre-processing now conditionally stores surrogate data based on this setting. Removes obsolete aerodynamic surrogate .pkl files from validation, and comments out most regression test modules except Boeing 787 payload range validation.

ssankalp26 committed:
Version update

ARmolloy committed:
Update aerodynamics.py

mclarke2 commented on RCAIDE/Library/Methods/Performance/compute_payload_range_diagram.py:
```diff
@@ -107,12 +108,16 @@ def compute_payload_range_diagram(mission = None, cruise_segment_tag = "cruise",
             payload_range  =  electric_payload_range_diagram(vehicle,mission,cruise_segment_tag,plot_diagram)
+    
+    for fname in os.listdir(os.path.dirname(os.path.abspath(sys.argv[0]))):
+        if fname.endswith(".pkl"):
```
Do not use pickle files. Store in CSV format.

mclarke2 commented on RCAIDE/Library/Methods/Performance/compute_payload_range_diagram.py:
```diff
@@ -107,12 +108,16 @@ def compute_payload_range_diagram(mission = None, cruise_segment_tag = "cruise",
         else:
             payload_range  =  electric_payload_range_diagram(vehicle,mission,cruise_segment_tag,plot_diagram)
+    
+    for fname in os.listdir(os.path.dirname(os.path.abspath(sys.argv[0]))):
```
No pickle format

mclarke2 commented on RCAIDE/Library/Mission/Common/Pre_Process/aerodynamics.py:
```diff
@@ -64,25 +67,34 @@ def aerodynamics(mission):
-            if segment.analyses.aerodynamics != None:
-                if last_tag!=  None:
-                    if segment.analyses.aerodynamics.settings.unique_segment_surrogate:
+        filename = os.path.join(os.path.dirname(os.path.abspath(sys.argv[0])), mission.tag + "_" + segment.tag + "_aerodynamic_surrogate.pkl")
```
No pickle, change to CSV

mclarke2 commented on RCAIDE/Library/Mission/Common/Pre_Process/aerodynamics.py:
```diff
@@ -64,25 +67,34 @@ def aerodynamics(mission):
+                            pickle.dump(segment.analyses.aerodynamics, file) 
+        else:
+            with open(filename, 'rb') as file:
+                segment.analyses.aerodynamics = pickle.load(file)
```
you should just store the coefficients, not the entire analyses. That's a lot of data, including processes, etc.

mclarke2 commented on RCAIDE/Library/Mission/Common/Pre_Process/aerodynamics.py:
```diff
@@ -64,25 +67,34 @@ def aerodynamics(mission):
+                        last_tag = tag  
+                    if segment.analyses.aerodynamics.settings.store_surrogate_data:
+                        with open(filename, 'wb') as file:
+                            pickle.dump(segment.analyses.aerodynamics, file) 
```
Instead of saving the whole analysis data structure just to get a surrogate, save the raw data used to create the surrogate. That way we can do UQ on our aircraft simulations down the road

ssankalp26 committed:
Add aerodynamic surrogate building to analyses

Introduced build_aerodynamic_surrogate methods to Athena_Vortex_Lattice and Vortex_Lattice_Method classes, enabling surrogate model construction. Updated aerodynamics pre-processing to use deep copies and improved surrogate data handling with pickle, ensuring surrogates are rebuilt after loading training data.

ssankalp26 commented:
Saving training data now instead of the surrogates.

ARmolloy committed:
Update turbofan_network_test.py

Landing configuration has extended landing gear and flaps deployed while the other segments (all other segments use "cruise" or "base" except for one with "reverse_thrust") do not. The effect of flaps on lift is calculated in the VLM, therefore reusing the VLM training data from a flapless configuration is causing issues I think

ARmolloy committed:
Update empirical_jet_noise_test.py

ssankalp26 commented on RCAIDE/Library/Mission/Common/Pre_Process/aerodynamics.py:
```diff
@@ -11,6 +11,9 @@
 import sys
+from copy import deepcopy
+
+from RCAIDE.Library.Methods.Aerodynamics.Vortex_Lattice_Method.build_VLM_surrogates import build_surrogate
```
Can be removed

mclarke2 committed:
updates

ssankalp26 committed:
Revert "Update turbofan_network_test.py"

This reverts commit 3d6476db624095e5496625277e74cf78412c6faa.

ssankalp26 committed:
Revert "Update empirical_jet_noise_test.py"

This reverts commit 627d407707b5f7f11332a351aaecedb27da3cd0e.

ssankalp26 committed:
Change location of where how the pickle files are saved for aerodynamic training data

Replaces 'store_surrogate_data' with 'store_training_data' across aerodynamics modules for clarity. Updates file naming conventions and related logic to consistently use 'training data' terminology.

ssankalp26 committed:
Remove unused build_aerodynamic_surrogate method

Deleted the build_aerodynamic_surrogate method from Athena_Vortex_Lattice as it was not being used. This helps clean up the code and improve maintainability.

ssankalp26 committed:
Fix incorrect reference in docstring for Liquid_Hydrogen

Updated the 'See Also' section in the docstring to reference the correct module path for load_hydrogen_properties, changing from Solids to Propellants.

ARmolloy committed:
Aircraft Aerodynamic Analysis Update

Added a dummy filepath to aircraft aerodynamic analysis
